sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java

```diff
@@ -215,5 +233,50 @@ private ByteBufFlux bodyIntern() {
         Connection internConnection() {
             return reactorNettyConnection;
         }
+
+        private void updateSubscriptionState(Subscription subscription) {
+            if (this.state.compareAndSet(ReactorNettyResponseState.NOT_SUBSCRIBED, ReactorNettyResponseState.SUBSCRIBED)) {
+                return;
+            }
+            // https://github.com/reactor/reactor-netty/issues/503
+            // FluxReceive rejects multiple subscribers, but not after a cancel().
+            // Subsequent subscribers after cancel() will not be rejected, but will hang instead.
+            // So we need to reject ones in cancelled state.
+            if (this.state.get() == ReactorNettyResponseState.CANCELLED) {
+                throw new IllegalStateException(
+                    "The client response body has been released already due to cancellation.");
+            }
+        }
+
+        /**
+         * Called by {@link ReactorNettyClient} when a cancellation is detected
+         * but the content has not been subscribed to. If the subscription never
+         * materializes then the content will remain not drained. Or it could still
+         * materialize if the cancellation happened very early, or the response
+         * reading was delayed for some reason.
+         */
+        private void releaseAfterCancel(HttpMethod method) {
+            if (mayHaveBody(method) && this.state.compareAndSet(ReactorNettyResponseState.NOT_SUBSCRIBED, ReactorNettyResponseState.CANCELLED)) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Releasing body, not yet subscribed");
+                }
+                this.bodyIntern()
+                    .doOnNext(byteBuf -> {})
+                    .subscribe(byteBuf -> {}, ex -> {});
+            }
+        }
+
+        private boolean mayHaveBody(HttpMethod method) {
+            int code = this.statusCode();
+            return !((code >= 100 && code < 200) || code == 204 || code == 205 ||
```

Comment:
Do 304 classified as non body status code ?

Reply:
Yes, 304 will be a non body status code

Reply:
So lets add that here